docs/csharp/tutorials/string-interpolation.md

```diff
@@ -1,122 +1,98 @@
-You’ll need to install your favorite code editor. The descriptions below
-use [Visual Studio Code](https://code.visualstudio.com/) which is an open
-source, cross platform editor. However, you can use whatever tools you are
-comfortable with.
+At compile time an interpolated string is typically transformed to the <xref:System.String.Format%2A?displayProperty=nameWithType> method call. That makes all the capabilities of the [string composite formatting](../../standard/base-types/composite-formatting.md) feature available to you to use with interpolated strings as well. If an interpolated string is implicitly converted to either <xref:System.FormattableString?displayProperty=nameWithType> or <xref:System.IFormattable?displayProperty=nameWithType>, at compile time it's transformed to the expression that produces a <xref:System.FormattableString> instance. That allows you to provide the <xref:System.IFormatProvider> implementation to be used during the resolution of an interpolated string into the result string.
 
-## Create the Application
+## How to specify a format string for an interpolated expression
 
-Now that you've installed all the tools, create a new .NET Core
-application. To use the command line generator, create a directory for your project, such as `interpolated`, and execute the following command in your favorite shell:
+You specify a format string that is supported by the type of the expression result by following the interpolated expression with a colon (":") and the format string:
 
 ```
-dotnet new console
+{<interpolatedExpression>:<formatString>}
 ```
 
-This command creates a barebones .NET Core project with a project file, *interpolated.csproj*, and a source code file, *Program.cs*. You will need to execute `dotnet restore` to restore the dependencies needed to compile this project.
+The following example shows how to specify standard and custom format strings for expressions that produce date and time or numeric results:
 
-[!INCLUDE[DotNet Restore Note](~/includes/dotnet-restore-note.md)]
+[!code-csharp-interactive[format string example](~/samples/snippets/csharp/tutorials/string-interpolation/Program.cs#2)]
 
-To execute the program, use `dotnet run`. You should see "Hello, World" output to the console.
+For more information, see the [Format String Component](../../standard/base-types/composite-formatting.md#format-string-component) section of the [Composite Formatting](../../standard/base-types/composite-formatting.md) topic. That section provides links to the topics that describe standard and custom format strings supported by .NET base types.
 
+## How to control the field width and alignment of the formatted interpolated expression
 
+You specify the minimum field width in which string representation of the expression result is inserted and the text alignment within that field by following the interpolated expression with a comma (",") and the constant expression that defines alignment:
```

Comment:
- Somewhat more concise: "You specify the minimum field width and the alignment of the string produced by the interpolated expression by following...
- the constant expression --> an expression (constant here is capable of misinterpretation)
"that defines alignment" is unnecessary, since you've already said it defines alignment and field width.

Reply:
>the constant expression --> an expression (constant here is capable of misinterpretation)

How can it be misinterpreted? I then try to clarify that. Just *an expression* is not correct: an expression must be possible to evaluate at the compile time. Thus I've used *constant* term.

Reply:
The confusion about constant is that it's typically interpreted as a fixed (not variable) value. Your example, though, uses variables to define field width.

Reply:
In my example, they are not variables, they are local constants. I cannot change their values after declaration.